language_service/src/signature_help.rs

```diff
@@ -65,23 +67,31 @@ impl<'a> Visitor<'a> for SignatureHelpFinder<'a> {
                                 self.compilation.unit.ast.names.get(path.id)
                             {
                                 if let (Some(item), _) = find_item(self.compilation, item_id) {
-                                    if let qsc::hir::ItemKind::Callable(callee) = &item.kind {
+                                    if let hir::ItemKind::Callable(callee) = &item.kind {
                                         // Check that the callee has parameters to give help for
                                         if !matches!(&callee.input.kind, hir::PatKind::Tuple(items) if items.is_empty())
                                         {
+                                            let documentation = parse_doc_for_summary(&item.doc);
+                                            let documentation = (!documentation.is_empty())
+                                                .then_some(documentation);
+
                                             let sig_info = SignatureInformation {
                                                 label: self
                                                     .display
                                                     .hir_callable_decl(callee)
                                                     .to_string(),
```

Comment:
Style: the indentation here is getting pretty wild... this is how the diff shows up on my laptop:

![image](https://github.com/microsoft/qsharp/assets/16928427/8d753c76-a4e0-45ca-94a1-84d830ffe040)

Would it be possible to extract at least part of this function to a helper function?

Reply:
I broke out the processing of direct callee expressions in anticipation of eventually supporting indirect callee expressions (and how that will likely look very different).